File: leetcode75/java/Leetcode345.java
package leetcode75.java;

//345. Reverse Vowels of a String
//https://leetcode.com/problems/reverse-vowels-of-a-string/description/

public class Leetcode345 {

    private final String VOWELS = "aeiouAEIOU";

    public String reverseVowels(String s) {
        char[] arr = s.toCharArray();
        int left = 0, right = arr.length - 1;

        while (left < right) {
            while (left < right && !VOWELS.contains("" + arr[left])) {
                left++;
            }
            while (left < right && !VOWELS.contains("" + arr[right])) {
                right--;
            }

            char temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;

            left++;
            right--;
        }

        return new String(arr);
    }

    public static void main(String[] args) {
        String s = "IceCreAm";
        Leetcode345 leetcode345 = new Leetcode345();
        String res = leetcode345.reverseVowels(s);
        System.out.println(res);
    }

}

File: leetcode75/python/leetcode345.py
#345. Reverse Vowels of a String
# https://leetcode.com/problems/reverse-vowels-of-a-string/description/

class Solution:

    VOWELS = 'aeiouAEIOU'

    def reverseVowels(self, s):
        listOfChar = list(s);

        left, right = 0, len(s)-1

        while left < right:

            while left < right and listOfChar[left] not in self.VOWELS:
                left += 1
            
            while left < right and listOfChar[right] not in self.VOWELS:
                right -= 1

            listOfChar[left], listOfChar[right] = listOfChar[right], listOfChar[left]
            left += 1
            right -= 1
        
        return ''.join(listOfChar)
    
if __name__ == '__main__':
    s = "IceCreAm"
    solution = Solution()
    print(solution.reverseVowels(s))